### PR TITLE
Add simnet model architecture

### LINKED ISSUE #96: Try using Simnet architecture
I found this architecture 'Simnet' by Amazon Development Services, it uses a variation of Siamese network, they use 2 extra shallower CNN models trained on downsampled images alongside the ImageNet, 
The final results are better.
https://arxiv.org/pdf/1709.08761.pdf
image

This would be a great experiment as per issue #1

### PR DESCRIPTION
**Pull request by Trion129**
Fixes #96.

Added a function to build Simnet.

To do:
- [x] Run and compare to general VGG16
- [x] Optimize L2_Regularization

I couldn't run it because of some issues with files missing from my data folder, would file those as separate issues.

### CODE REVIEW

**Comment by Trion129** [comment]
## Simnet
L2 = 0.005

```
Epoch 1/50
718/717 [==============================] - 207s 289ms/step - loss: 0.2769 - accuracy: 0.5276
Epoch 2/50
718/717 [==============================] - 207s 289ms/step - loss: 0.2200 - accuracy: 0.6421
Epoch 3/50
718/717 [==============================] - 197s 274ms/step - loss: 0.2522 - accuracy: 0.5275
Epoch 4/50
718/717 [==============================] - 192s 267ms/step - loss: 0.1558 - accuracy: 0.7950
Epoch 5/50
718/717 [==============================] - 212s 296ms/step - loss: 0.1402 - accuracy: 0.8250
Epoch 6/50
718/717 [==============================] - 202s 282ms/step - loss: 0.1851 - accuracy: 0.6920
Epoch 7/50
718/717 [==============================] - 192s 268ms/step - loss: 0.2250 - accuracy: 0.6108
```

## VGG16

```
Epoch 1/50
750/749 [==============================] - 175s 234ms/step - loss: 0.2110 - accuracy: 0.6640
Epoch 2/50
750/749 [==============================] - 187s 249ms/step - loss: 0.1734 - accuracy: 0.7834
Epoch 3/50
750/749 [==============================] - 181s 241ms/step - loss: 0.2327 - accuracy: 0.6308
Epoch 4/50
750/749 [==============================] - 183s 244ms/step - loss: 0.1570 - accuracy: 0.7752
Epoch 5/50
750/749 [==============================] - 175s 234ms/step - loss: 0.2410 - accuracy: 0.5894
Epoch 6/50
750/749 [==============================] - 177s 235ms/step - loss: 0.2076 - accuracy: 0.6828
Epoch 7/50
750/749 [==============================] - 174s 233ms/step - loss: 0.2124 - accuracy: 0.6586
Epoch 8/50
750/749 [==============================] - 169s 226ms/step - loss: 0.1688 - accuracy: 0.7575
```

**Comment by Trion129** [comment]
For 0.01, the loss stayed 0.17+ so 0.005 seems fine for simnetlike.
```
Epoch 1/50
730/729 [==============================] - 185s 254ms/step - loss: 0.2689 - accuracy: 0.5552
Epoch 2/50
730/729 [==============================] - 182s 250ms/step - loss: 0.1877 - accuracy: 0.7043
Epoch 3/50
730/729 [==============================] - 184s 251ms/step - loss: 0.1790 - accuracy: 0.6996
Epoch 4/50
730/729 [==============================] - 179s 246ms/step - loss: 0.2180 - accuracy: 0.6673
Epoch 5/50
730/729 [==============================] - 180s 247ms/step - loss: 0.2017 - accuracy: 0.6911
```

**Comment by marco-c** [comment]
@Trion129 is this ready for review?

**Comment by Trion129** [comment]
@marco-c Yeah, although I didn't train the main simnet architecture to verify it really performs better (lack of processing power), just experimented with the simnet-like model and it performs slightly better than vgg-like model! 0.1402 vs 0.1570
L2_regularization that worked best for simnetlike is 0.005

**Comment by marco-c** [comment]
Could you resolve conflicts?

**Comment by marco-c** [comment]
There are some linting issues.

File: autowebcompat/network.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def create_vgg16_network(input_shape):
     x = MaxPooling2D(pool_size=(2, 2), strides=(2, 2))(x)
 
     # Block 4
-    x = Conv2D(512, (3, 3), activation='relu', padding='same')(x)
+    x = Conv2D(512, (3, 3), flake8activation='relu', padding='same')(x)
```
Comment (marco-c) [inline autowebcompat/network.py]: I suppose there's something wrong here :smile:

Comment (Trion129) [reply autowebcompat/network.py]: :P Whoa! how did that happen

**Comment by Trion129** [comment]
weird, I don't have the isort error locally

**Comment by marco-c** [comment]
Can you also add simnet to the list of supported networks? At the beginning of the file.

**Pull request merged**